.github/workflows/pytest-core-nompi.yml

```diff
@@ -11,127 +11,128 @@ on:
       - master
 
 jobs:
-  ci-linux:
+  pytest:
     name: ${{ matrix.name }}
-    runs-on: ubuntu-16.04
+    runs-on: "${{ matrix.os }}"
 
     env:
-      DEVITO_ARCH: "gcc-${{ matrix.gcc-version }}"
+      DEVITO_ARCH: "${{ matrix.arch }}"
       DEVITO_OPENMP: ${{ matrix.openmp }}
       DEVITO_BACKEND: "core"
       OMP_NUM_THREADS: 2
-      CC: "gcc-${{ matrix.gcc-version }}"
-      CXX: "g++-${{ matrix.gcc-version }}"
       PYTHON_VERSION: "${{ matrix.python-version }}"
+      TESTS: "tests/"
 
     strategy:
       # Prevent all build to stop if a single one fails
       fail-fast: false
 
       matrix:
         name: [
-           py36-gcc49-openmp,
-           py37-gcc5-openmp,
-           py38-gcc6-openmp,
-           py36-gcc7-openmp,
-           py37-gcc7-noopenmp,
-           py37-gcc8-openmp,
-           py38-gcc9-openmp
+           pytest-ubuntu-py36-gcc49-omp,
+           pytest-ubuntu-py37-gcc5-omp,
+           pytest-ubuntu-py38-gcc6-omp,
+           pytest-ubuntu-py36-gcc7-omp,
+           pytest-ubuntu-py37-gcc7-noomp,
+           pytest-ubuntu-py37-gcc8-omp,
+           pytest-ubuntu-py38-gcc9-omp,
+           pytest-osx-py37-clang-omp,
+           pytest-docker-py36-gcc-omp
         ]
         include:
-            - name: py36-gcc49-openmp
-              python-version: 3.6
-              gcc-version: 4.9
-              openmp: 1
-
-            - name: py37-gcc5-openmp
-              python-version: 3.7
-              gcc-version: 5
-              openmp: 1
-
-            - name: py38-gcc6-openmp
-              python-version: 3.8
-              gcc-version: 6
-              openmp: 1
-
-            - name: py36-gcc7-openmp
-              python-version: 3.6
-              gcc-version: 7
-              openmp: 1
-
-            - name: py37-gcc7-noopenmp
-              python-version: 3.7
-              gcc-version: 7
-              openmp: 0
-
-            - name: py37-gcc8-openmp
-              python-version: 3.7
-              gcc-version: 8
-              openmp: 1
-
-            - name: py38-gcc9-openmp
-              python-version: 3.8
-              gcc-version: 9
-              openmp: 1
+        - name: pytest-ubuntu-py36-gcc49-omp
```

Comment:
Should the tests be in ascending order of python version? Now it is gcc, but just saying...